tests/installation/module_registration/register_extensions_and_modules.pm

```diff
@@ -11,12 +11,32 @@
 use base 'y2_installbase';
 use strict;
 use warnings;
-use testapi qw(save_screenshot get_required_var);
+use testapi qw(save_screenshot get_required_var get_var);
 
 sub run {
     my @scc_addons = grep($_, split(/,/, get_required_var('SCC_ADDONS')));
     $testapi::distri->get_module_registration()->register_extension_and_modules([@scc_addons]);
     save_screenshot;
+    my $timeout = 60 * get_var('TIMEOUT_SCALE', 1);
+    $testapi::distri->wait_registration_common_regcode_finished({timeout => $timeout, interval => 2, message => 'Page to insert module registration code did not appear'});
+    foreach my $addon (@scc_addons) {
+        if ($addon eq 'we' && get_var('SCC_REGCODE_WE', '')) {
+            my $regcode = get_var('SCC_REGCODE_WE');
+            $testapi::distri->get_module_regcode()->add_separate_we_registration_code($regcode, $timeout);
+            save_screenshot;
+            $testapi::distri->get_module_regcode()->trust_gnupg_key();
+        }
+        if ($addon eq 'ha' && get_var('SCC_REGCODE_HA', '')) {
+            my $regcode = get_var('SCC_REGCODE_HA');
+            $testapi::distri->get_module_regcode()->add_separate_ha_registration_code($regcode, $timeout);
+            save_screenshot;
+        }
+        if ($addon eq 'ltss' && get_var('SCC_REGCODE_LTSS', '')) {
+            my $regcode = get_var('SCC_REGCODE_LTSS');
+            $testapi::distri->get_module_regcode()->add_separate_ltss_registration_code($regcode, $timeout);
+            save_screenshot;
```

Comment:
you don't need the final screenshot at the end of previous two `if` because it is taken by the test module in post_run_hook.